src/drive/mobile/containers/FirstUploadModal.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react'
+import Modal, { ModalContent } from 'cozy-ui/react/Modal'
+import { connect } from 'react-redux'
+import withPersistentState from '../lib/withPersistentState'
+import styles from '../styles/uploadprogression'
+import { isIos } from '../lib/device'
+
+const MINIMUM_LONG_UPLOAD_FILES_COUNT = 50
+
+class FirstUploadModal extends Component {
+  render() {
+    const { onClose } = this.props
+    const { t } = this.context
+
+    return (
+      <Modal title={t('mobile.first_sync.title')} withCross={false}>
+        <ModalContent>
+          <h4>{t('mobile.first_sync.tips')}</h4>
+          <ul>
+            <li>{t('mobile.first_sync.tip_wifi')}</li>
+            <li>{t('mobile.first_sync.tip_bed')}</li>
+            {isIos() && <li>{t('mobile.first_sync.tip_lock')}</li>}
+          </ul>
+          <p>{t('mobile.first_sync.result')}</p>
+          <button onClick={onClose} className={styles['btn--full-width']}>
+            {t('mobile.first_sync.button')}
+          </button>
+        </ModalContent>
+      </Modal>
+    )
+  }
+}
+
+const showOnlyOnce = WrappedComponent => {
+  return class extends Component {
+    state = {
+      shown: false
+    }
+
+    markAsShown = () => {
+      this.setState({ shown: true })
+    }
+
+    render() {
+      const { uploading } = this.props
+      const { shown } = this.state
+
+      if (uploading && !shown)
+        return <WrappedComponent {...this.props} onClose={this.markAsShown} />
+      else return null
+    }
+  }
+}
+
+const OnlyOnceFirstUploadModal = withPersistentState(
+  showOnlyOnce(FirstUploadModal),
+  'OnlyOnceFirstUploadModal'
+)
+
+const mapStateToProps = state => ({
+  uploading:
+    state.mobile.mediaBackup.uploading &&
+    state.mobile.mediaBackup.currentUpload &&
+    state.mobile.mediaBackup.currentUpload.messageData &&
+    state.mobile.mediaBackup.currentUpload.messageData.total_upload &&
+    state.mobile.mediaBackup.currentUpload.messageData.total_upload >=
+      MINIMUM_LONG_UPLOAD_FILES_COUNT
+})
```

Comment:
It is really hard to catch the intention here. `uploading` is a boolean, its value is `true` if and only if all other booleans are `true`. All other booleans are:
- `uploading`: ok, so to be `true` local `uploading` needs global `uploading` to be true ;
- `currentUpload`... I guess if `currentUpload` is set (surely not as a boolean), global `uploading` is true by the way ;
- `messageData`... its a `currentUpload`'s property... ohhhh, I see something !

In fact, what we want for local `uploading` to be true is only two things:
1. global `uploading` to be true
1. `total_upload` to be greater than `MINIMUM_LONG_UPLOAD_FILES_COUNT`

but as `total_upload` is a subsubsubproperty, we need to check the existence of parent properties. Destructuring is a very cool feature for such a thing:

```js
const mapStateToProps = state => {
    const { uploading = false, currentUpload = { messageData: { total_upload: 0 } } } = obj2.state.mobile.mediaBackup
    const { messageData: { total_upload } } = currentUpload
    return {
        uploading: uploading && total_upload >= MINIMUM_LONG_UPLOAD_FILES_COUNT
    }
}
```

Reply:
That is cool, I didn't knew you could destructure with default values. 👍